test/c/test_sabre_layout.c

```diff
@@ -0,0 +1,325 @@
+            qk_circuit_instruction_clear(expected_inst);
+            free(res_inst);
+            free(expected_inst);
+            return false;
+        }
+        for (uint32_t j = 0; j < res_inst->num_clbits; j++) {
+            if (res_inst->clbits[j] != expected_inst->clbits[j]) {
+                printf("Clbit %d for gate %d are different %d was found and expected %d", j, i,
+                       res_inst->clbits[j] != expected_inst->clbits[j]);
+                qk_circuit_instruction_clear(res_inst);
+                qk_circuit_instruction_clear(expected_inst);
+                free(res_inst);
+                free(expected_inst);
+                return false;
+            }
+        }
+        if (res_inst->num_params != expected_inst->num_params) {
+            printf("Gate %d have different number of params %d was found and expected %d", i,
+                   res_inst->num_params, expected_inst->num_params);
+            qk_circuit_instruction_clear(res_inst);
+            qk_circuit_instruction_clear(expected_inst);
+            free(res_inst);
+            free(expected_inst);
+            return false;
+        }
+        for (uint32_t j = 0; j < res_inst->num_params; j++) {
+            if (res_inst->params[j] != expected_inst->params[j]) {
+                printf("Parameter %d for gate %d are different %d was found and expected %d", j, i,
+                       res_inst->params[j] != expected_inst->params[j]);
+                qk_circuit_instruction_clear(res_inst);
+                qk_circuit_instruction_clear(expected_inst);
+                free(res_inst);
+                free(expected_inst);
+                return false;
+            }
+        }
+        qk_circuit_instruction_clear(res_inst);
+        qk_circuit_instruction_clear(expected_inst);
+    }
+    free(res_inst);
+    free(expected_inst);
+    return true;
+}
+
+/**
+ * Test running sabre layout that requires layout and routing
+ */
+int test_sabre_layout_applies_layout(void) {
+    int result = Ok;
+
+    const uint32_t num_qubits = 5;
+    QkTarget *target = qk_target_new(num_qubits);
+
+    qk_target_add_instruction(target, qk_target_entry_new(QkGate_U));
+    QkTargetEntry *cx_entry = qk_target_entry_new(QkGate_CX);
+    for (uint32_t i = 0; i < num_qubits - 1; i++) {
+        uint32_t qargs[2] = {i, i + 1};
+        double inst_error = 0.0090393 * (num_qubits - i);
+        double inst_duration = 0.020039;
+        qk_target_entry_add_property(cx_entry, qargs, 2, inst_duration, inst_error);
+    }
+
+    qk_target_add_instruction(target, cx_entry);
+
+    QkCircuit *qc = qk_circuit_new(5, 0);
+    for (uint32_t i = 0; i < qk_circuit_num_qubits(qc) - 1; i++) {
+        uint32_t qargs[2] = {0, i + 1};
+        qk_circuit_gate(qc, QkGate_CX, qargs, NULL);
+    }
+    QkTranspileLayout *layout_result =
+        qk_transpiler_pass_standalone_sabre_layout(qc, target, 4, 20, 20, 2025);
+
+    QkOpCounts op_counts = qk_circuit_count_ops(qc);
```

Comment:
I don't think we ever free this.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14711/commits/6d95b87cbf14fba5993c6dc4390ca68dffd1f83e